File: Chapter3.CoffeeShop.Java/src/Beverages/Beverage.java
package Beverages;

public abstract class Beverage {
    String description = "Unknown Beverage";
    Size size = Size.GRANDE;
    public enum Size { TALL, GRANDE, VENTI};
    public String getDescription() {
        return description;
    }

    public void setSize(Size size) {
        this.size = size;
    }
    public Size getSize() {
        return size;
    }

    public abstract double cost();
}

File: Chapter3.CoffeeShop/Models/Beverages/Beverage.cs
namespace Chapter3.CoffeeShop.Models;

public abstract class Beverage
{
    public enum Size
    {
        Tall, Grande, Venti
    }

    private Size _size = Size.Tall;
    protected string _description = "Unknown Beverage";
    public virtual string GetDescription() => _description;
    public abstract double Cost();
    public void SetSize(Size size) => _size = size;
    public virtual Size GetSize() => _size;
}